shared/models.go

```diff
@@ -40,3 +44,56 @@ type Browser struct {
 type Token struct {
 	Secret string `json:"secret"`
 }
+
+type Manifest struct {
+	Items   ManifestItems `json:"items,omitempty"`
+	Version *int          `json:"version,omitempty"`
+}
+
+func (m Manifest) FilterByPath(paths mapset.Set) (result Manifest, err error) {
+	if result.Items.Manual, err = m.Items.Manual.FilterByPath(paths); err != nil {
+		return result, err
+	}
+	if result.Items.Reftest, err = m.Items.Reftest.FilterByPath(paths); err != nil {
+		return result, err
+	}
+	if result.Items.TestHarness, err = m.Items.TestHarness.FilterByPath(paths); err != nil {
+		return result, err
+	}
+	return result, nil
+}
+
+type ManifestItems struct {
+	Manual      *ManifestItem `json:"manual,omitempty"`
+	Reftest     *ManifestItem `json:"reftest,omitempty"`
+	TestHarness *ManifestItem `json:"testharness,omitempty"`
```

Comment:
`ManifestItem` is a map, right? [Maps are already references](https://golang.org/doc/faq#references), so we don't need to make a pointer of it again.

Reply:
Done.